shards/modules/gfx/shader/core_shards.hpp

```diff
@@ -409,6 +420,34 @@ template <typename TShard> struct Read final : public IOBase {
   }
 };
 
+struct ShaderReadInput : public Read<blocks::ReadInput> {
+  static SHOptionalString help() {
+    return SHCCSTR("This shard either reads the value of the specified vertex attribute supplied to the vertex or reads the interpolated value of that attribute, output from the vertex stage and provided to the pixel.");
```

Comment:
Might be useful to clearly state in what scenario what happens, e.g.
"In the vertex stage, this will read mesh vertex attributes"
"In the fragment stage this will read the interpolated output from the vertex stage"